### DISCUSSION TITLE
Need to get the current rules state in my application

### DISCUSSION THREAD

**Discussion by esamaral** [react-querybuilder/react-querybuilder#768] [Q&A] [answered]
**Description of the feature**
I need to persist the rules that are added to the query builder in my component, but i did not find a way to get that state from the querybuilder component.
Theres someway that this can be done?

**Comment by jakeboone02** [comment, marked as answer]
The [`onQueryChange`](https://react-querybuilder.js.org/docs/components/querybuilder#onquerychange) prop accepts a function that will be called with the latest query object each time the query is updated. In the function you provide, you can accept the query value and store/persist it any way you like.
